src/ec/packages/ec-component-ecl-compliance/components/ecl-compliance-media-container.html.twig

```diff
@@ -0,0 +1,56 @@
+{#
+  Parameters:
+    alt (string),
+    image (string),
+    embedded_media: (string),
+    sources: (array of objects),
+    tracks: (array of objects)
+#}
+
+{% extends "@ecl-twig/ec-component-ecl-compliance/ecl-compliance.html.twig" %}
+
+{% set message = '' %}
+{% set component = 'media-container' %}
+{% set no_embedded_media_case = embedded_media is not defined or embedded_media is empty %}
+{% set no_sources_case = sources is not defined or sources is empty or sources is not iterable %}
+{% set no_tracks_case = tracks is not defined or tracks is empty or tracks is not iterable %}
+{% set no_video_resources_case = no_sources_case and no_tracks_case %}
+{% set no_image_case = image is not defined or image is empty %}
+{% set no_alt_case = no_image_case == false and (alt is not defined or alt is empty) %}
+
+{# no media provided at all #}
+{% set not_compliant = no_image_case and no_embedded_media_case and no_video_resources_case %}
+{% if not_compliant %}
+  {% set message = 'One media is required for the media-container component, an "image", an "embedded_media" or video "tracks" and "sources".' %}
+{% endif %}
+{# No embedded_media, we check for video sources #}
+{% if no_embedded_media_case and no_sources_case == false %}
+  {% for source in sources %}
+    {% if (source.src is not defined or source.src is empty) or (source.type is not defined or source.type is empty) %}
+      {% set not_compliant = true %}
+    {% endif %}
+  {% endfor %}
+  {% if not_compliant %}
+    {% set message = message ~ 'The "sources" parameter should be an array with a "src" and a "type" for each element.' %}
```

Comment:
As you are iterating through `sources` for the compatibility check, you can (I think) allow yourself to use this `message` variable to include more specific information which source exactly is the problematic.

Reply:
it's true, we can clearly be more granular, but the evaluation i made is that this was the right "balance", between the logic to be set, the messages to be prepared and the "value" in term of information provided to the user. Clearly we could also specifically mention the element or th elements in a loop that are causing the failure, but this is a tool for developers, we aim to help, but i think it's enough if you see that your problem is coming from a certain array where we tell you which are the needed keys.
In this case for instance you have sources but also tracks, the number of different use cases would increase a lot, that's the reason for this "limitation".
Anyway, this approach has been used many times, in many loops, therefore i would in case considering a second iteration on this, if we want to, my personal opinion is that is not worth but let's decide it together :)

Reply:
`this is a tool for developers` if this is true, why not making a CLI for these checks?